services/provider/server/server.go

```diff
@@ -412,6 +418,18 @@ func validateTicket(ticket string, pubKey *rsa.PublicKey) error {
 	if err != nil {
 		return fmt.Errorf("failed to decode payload. %v", err)
 	}
+
+	var ticketDate TicketDate
+	err = json.Unmarshal(message, &ticketDate)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal message. %v", err)
+	}
+
+	if ticketDate.ExpirationDate < time.Now().Unix() {
+		return fmt.Errorf("%s id %d date is lesser than the current date", ticketDate.ID, ticketDate.ExpirationDate)
+	}
+	klog.Infof("onboarding ticket %s has been verified successfully", ticketDate.ID)
+
 	signature, err := base64.StdEncoding.DecodeString(ticketArr[1])
 	if err != nil {
 		return fmt.Errorf("failed to decode signature. %v", err)
```

Comment:
Incorporate id into message, maybe `failed to decode onboarding ticket (<id>) signature: %v`